apps/s_cb.c

```diff
@@ -1355,3 +1356,51 @@ void ssl_ctx_security_debug(SSL_CTX *ctx, int verbose)
     SSL_CTX_set_security_callback(ctx, security_callback_debug);
     SSL_CTX_set0_security_ex_data(ctx, &sdb);
 }
+
+static void keylog_callback(const SSL *ssl, const char *line)
+{
+    if (bio_keylog == NULL) {
+        BIO_printf(bio_err, "Keylog callback is invoked without valid file!\n");
+        return;
+    }
+
+    /*
+     * There might be concurrent writers to the keylog file, so we must ensure
+     * that one line is written at once.
+     */
+    BIO_puts(bio_keylog, line);
```

Comment:
Possible, but then I expect that every caller will likely need to do the same. What about making sure that the line is actually terminated by a `\n`?